docs/walkthrough/index.md

```diff
@@ -0,0 +1,328 @@
+```
+
+where 200 is the number of messages sent.
+
+We can also look at the Prometheus server and check that it's pulling in metrics from Promitor.
+
+Cancel the previous port-forward command and run `kubectl port-forward svc/<prometheus-release-name>-prometheus-server 8080:80`.
+
+Now, if you check http://localhost:8080, you should be able to enter Prometheus queries.
+
+Query `demo_queue_size` and as long as all your pods are up and running and both Promitor and Prometheus have scraped metrics at least once, you should see a value that matches the number of messages in your queue.
+
+# Visualization and alerting
+
+## Install Grafana
+
+Grafana's chart has a few default values you may not want long term - persistant storage is disabled and admin username/password is randomly generated - but for our sample the out-of-the-box install will work. 
+
+Run `helm install stable/grafana --name grafana` and you should see output that includes this command:
+
+```
+kubectl get secret --namespace default grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
+```
+
+Run this to get your Grafana password.
+
+Now you can use `kubectl port-forward` again to log in to your Grafana dashboard. `kubectl port-forward svc/grafana 8080:80` will make your dashboard available at http://localhost:8080, and you can log in with username 'admin' and the password you retrieved.
+
+## Add Prometheus as a data source
+
+After logging in, you should see an option to "Add a Data Source." Click that, and choose the Prometheus source type (if it's not immediately visible, search for it).
+
+The only setting you should need to edit here is the URL, under the HTTP section. Within your cluster, `http://<prometheus-release-name>-prometheus-server.default.svc.cluster.local` should resolve to the Prometheus server service. (Default in that URL refers to the namespace - if you installed in a namespace other than default, change that.)
+
+Set your service's name as the Prometheus URL in Grafana, and save the data source. It should tell you that the data source is working.
+
+## Create a Grafana dashboard for queue metrics
+
+First, we'll make a basic dashboard with the metric we set up in Promitor. Then you can use a pre-built dashboard to show Prometheus' default Kubernetes metrics.
+
+Go to the + button on the sidebar, and choose "Dashboard". To make a simple graph showing your queue size, you can write `demo_queue_size` in the query field. Click out of that input field and the graph should update.
+
+To see more, you can go back to Service Bus Explorer and send or receive messages. Your Grafana graph won't update immediately, but you should see results in a few minutes. 
+
+In order to see results without manually refreshing, find the dropdown menu in the top right corner that sets the time range of the graph. Here you can edit time range and refresh rate.
+
+Make sure to save your new dashboard before exiting the page.
+
+## Import an existing Grafana dashboard
```

Comment:
This feels more like a `Creating a Kubernetes dashboard`, right? This is great, I'm just wondering if it really matters for this scenario?

We can keep it, but would then suggest changing the name of the section?

Reply:
We can remove it if you'd like, I just figured it would help users see the full breadth of options they had with Grafana, as the first graph they make is *very* simple. Can definitely kill it if you still don't think it's necessary :+1:

Reply:
Let's keep it in for now!